iocore/net/SSLNetVConnection.cc

```diff
@@ -1260,9 +1260,9 @@ SSLNetVConnection::sslServerHandShakeEvent(int &err)
       this->reenable(&read.vio);
       this->read.triggered = 1;
     }
-  }
+  } else
```

Comment:
Is this really ok? I didn't checked where `ssl_error` is changed, but it seems like `ssl_error` can be `SSL_ERROR_WANT_READ` at least. I guess we should update `err` in the case at minimum?

Reply:
Imaging a plugin hooked on SNI or Cert HOOK and set the vc to blind tunnel.

Reply:
Added the else so in the async case, the ssl_error == SSL_ERROR_WANT_ASYNC case wouldn't trigger the error debug message.  Probably doesn't matter much one way or the other.

Reply:
Skipping debug log for SSL_ERROR_WANT_ASYNC is reasonable. However, I don't understand why we can skip other parts as well. Maybe I just don't understand how async works, but here are my concerns:

Don't we need to update `err` so that the caller can handle errors? If it was SSL_ERROR_WANT_ASYNC then there would be nothing to do, but I guess other errors can happen inside the if clause.

Also, like oknet pointed out, there is code for blind tunnels. Can we say the code is not needed for async case?

Reply:
Ok, I see the issue for the blind tunnel.  And the err variable. Added another commit which removed the "else".  Other than the extra debug message, the else wasn't really fixing anything.